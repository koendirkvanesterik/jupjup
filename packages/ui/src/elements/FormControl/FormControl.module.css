.form-control {
  @apply flex flex-col;

  &__label,
  &__hint {
    @apply flex items-center justify-between;
    @apply px-1 py-2;
    @apply select-none;
  }

  &__label {
    @apply text-sm;
  }

  &__hint {
    @apply text-xs;

    &--error {
      @apply text-red-500;
    }
  }
}
