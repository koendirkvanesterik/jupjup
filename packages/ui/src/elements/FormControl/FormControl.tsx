import classNames from 'classnames'
import { PropsWithChildren, ReactNode } from 'react'
import styles from './FormControl.module.css'

type Props = PropsWithChildren<{
  error?: boolean
  hint?: ReactNode | string
  label?: ReactNode | string
}>

export const FormControl = ({ children, error, hint, label }: Props) => (
  <div className={styles['form-control']}>
    {label && <div className={styles['form-control__label']}>{label}</div>}
    {children}
    {hint && (
      <div
        className={classNames(
          styles['form-control__hint'],
          error && styles['form-control__hint--error'],
        )}
      >
        {hint}
      </div>
    )}
  </div>
)
