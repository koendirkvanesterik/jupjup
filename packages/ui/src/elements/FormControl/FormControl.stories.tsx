import { Input } from '../Input'
import { FormControl } from './FormControl'

export const TextInput = () => (
  <div className="max-w-xs">
    <FormControl
      hint={<a href="#">Forgot your e-mail address?</a>}
      label="E-mail address"
    >
      <Input placeholder="Type here" />
    </FormControl>
  </div>
)

export default {
  component: FormControl,
  title: 'Elements/FormControl',
}
