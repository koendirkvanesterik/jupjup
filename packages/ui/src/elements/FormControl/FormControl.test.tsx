import { render } from '@testing-library/react'
import { FormControl } from './FormControl'

it('should render as expected', () => {
  const { container } = render(<FormControl />)
  expect(container).toMatchSnapshot()
})
