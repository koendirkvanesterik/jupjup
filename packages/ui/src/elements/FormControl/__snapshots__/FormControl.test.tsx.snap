// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render as expected 1`] = `
<div>
  <div
    class="form-control"
  />
</div>
`;
